### ISSUE TITLE
Difficulties running evaluations: config entry in SETUP_RETRY state

### ISSUE THREAD

**Issue by domdomegg** [allenporter/home-assistant-datasets#49]
## My goal

I'm keen to understand how to effectively evaluate LLM agents that use tools, like being able to control Home Assistant. I'd like to be able to run the evaluations here, and maybe even help expand them + expand their use in common frameworks (e.g. [AISI inspect](https://inspect.ai-safety-institute.org.uk/))

To begin with, I'm trying to run the evaluations using ollama and llama3.2:1b.

## Problems

Unfortunately, when I run the evaluations I get an AssertionError, stemming from `mock_conversation_agent_config_entry` in `fixtures.py`. The assertion that fails is checking the config entry is in the `ConfigEntryState.LOADED` state. Printing it out, it appears to instead be in the `ConfigEntryState.SETUP_RETRY` state.

<details><summary>Full logs</summary>

```
―――――――――――――――――――――― ERROR at setup of test_assist_actions[llama3.2-1b-home1_us_smart_lock-unlock_all_the_doors-True-True] ――――――――――――――――――――――

request = <SubRequest 'conversation_agent_config_entry' for <Coroutine test_assist_actions[llama3.2-1b-home1_us_smart_lock-unlock_all_the_doors-True-True]>>
kwargs = {'hass': <HomeAssistant RUNNING>, 'model_config': ModelConfig(model_id='llama3.2-1b', domain='ollama', description="Me...'llama3.2:1b'}, config_entry_options={'llm_hass_api': 'assist', 'num_ctx': 8192}, version=None), 'system_prompt': None}
event_loop_fixture_id = 'event_loop', setup = <function _wrap_async_fixture.<locals>._async_fixture_wrapper.<locals>.setup at 0xffffb1e7bce0>

    @functools.wraps(fixture)
    def _async_fixture_wrapper(request: FixtureRequest, **kwargs: Any):
        func = _perhaps_rebind_fixture_func(fixture, request.instance)
        event_loop_fixture_id = _get_event_loop_fixture_id_for_async_fixture(
            request, func
        )
        event_loop = request.getfixturevalue(event_loop_fixture_id)
        kwargs.pop(event_loop_fixture_id, None)
    
        async def setup():
            res = await func(**_add_kwargs(func, kwargs, event_loop, request))
            return res
    
>       return event_loop.run_until_complete(setup())

/home/vscode/.local/lib/python3.12/site-packages/pytest_asyncio/plugin.py:366: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
/usr/local/lib/python3.12/asyncio/base_events.py:687: in run_until_complete
    return future.result()
/home/vscode/.local/lib/python3.12/site-packages/pytest_asyncio/plugin.py:363: in setup
    res = await func(**_add_kwargs(func, kwargs, event_loop, request))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

hass = <HomeAssistant RUNNING>
model_config = ModelConfig(model_id='llama3.2-1b', domain='ollama', description="Meta's Llama 3.2 goes small with 1B and 3B models.",....0.0.1:11434', 'model': 'llama3.2:1b'}, config_entry_options={'llm_hass_api': 'assist', 'num_ctx': 8192}, version=None)
system_prompt = None

    @pytest.fixture(name="conversation_agent_config_entry")
    async def mock_conversation_agent_config_entry(
        hass: HomeAssistant, model_config: ModelConfig, system_prompt: str | None
    ) -> MockConfigEntry | None:
        if model_config.domain == "homeassistant":
            return None
        options = {}
        if model_config.config_entry_options:
            options.update(model_config.config_entry_options)
        if system_prompt:
            options["prompt"] = system_prompt
        config_entry = MockConfigEntry(
            domain=model_config.domain,
            data=model_config.config_entry_data,
            options=options,
            version=model_config.version or 1,
        )
        config_entry.add_to_hass(hass)
        await hass.config_entries.async_setup(config_entry.entry_id)
>       assert config_entry.state == ConfigEntryState.LOADED
E       AssertionError

home_assistant_datasets/fixtures.py:136: AssertionError
-------------------------------------------------------------- Captured stderr setup --------------------------------------------------------------
INFO:homeassistant.loader:Loaded synthetic_home from custom_components.synthetic_home
WARNING:homeassistant.loader:We found a custom integration synthetic_home which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
INFO:homeassistant.loader:Loaded homeassistant from homeassistant.components.homeassistant
INFO:homeassistant.setup:Setting up homeassistant
INFO:homeassistant.loader:Loaded conversation from homeassistant.components.conversation
INFO:homeassistant.loader:Loaded http from homeassistant.components.http
INFO:homeassistant.loader:Loaded intent from homeassistant.components.intent
INFO:homeassistant.setup:Setting up http
WARNING:aiohttp_fast_zlib:zlib_ng and isal are not available, falling back to zlib, performance will be degraded.
INFO:homeassistant.setup:Setting up intent
INFO:homeassistant.setup:Setting up conversation
INFO:homeassistant.setup:Setting up synthetic_home
INFO:custom_components.synthetic_home.model:value=['climate.ClimateEntityFeature.FAN_MODE', 'climate.ClimateEntityFeature.TURN_ON', 'climate.ClimateEntityFeature.TURN_OFF', 'climate.ClimateEntityFeature.TARGET_TEMPERATURE_RANGE'] (type=<class 'list'>)
INFO:custom_components.synthetic_home.model:value=['vacuum.VacuumEntityFeature.START', 'vacuum.VacuumEntityFeature.STOP', 'vacuum.VacuumEntityFeature.STATE', 'vacuum.VacuumEntityFeature.PAUSE', 'vacuum.VacuumEntityFeature.RETURN_HOME'] (type=<class 'list'>)
INFO:custom_components.synthetic_home.model:value=['media_player.MediaPlayerEntityFeature.PLAY', 'media_player.MediaPlayerEntityFeature.PAUSE', 'media_player.MediaPlayerEntityFeature.STOP', 'media_player.MediaPlayerEntityFeature.VOLUME_STEP', 'media_player.MediaPlayerEntityFeature.VOLUME_MUTE', 'media_player.MediaPlayerEntityFeature.VOLUME_SET', 'media_player.MediaPlayerEntityFeature.TURN_ON', 'media_player.MediaPlayerEntityFeature.TURN_OFF', 'media_player.MediaPlayerEntityFeature.NEXT_TRACK', 'media_player.MediaPlayerEntityFeature.PREVIOUS_TRACK'] (type=<class 'list'>)
INFO:custom_components.synthetic_home.model:value=['cover.CoverEntityFeature.OPEN', 'cover.CoverEntityFeature.CLOSE'] (type=<class 'list'>)
INFO:custom_components.synthetic_home.model:value=['cover.CoverEntityFeature.OPEN', 'cover.CoverEntityFeature.CLOSE', 'cover.CoverEntityFeature.SET_POSITION'] (type=<class 'list'>)
INFO:homeassistant.loader:Loaded binary_sensor from homeassistant.components.binary_sensor
INFO:homeassistant.loader:Loaded climate from homeassistant.components.climate
INFO:homeassistant.loader:Loaded fan from homeassistant.components.fan
INFO:homeassistant.loader:Loaded light from homeassistant.components.light
INFO:homeassistant.loader:Loaded cover from homeassistant.components.cover
INFO:homeassistant.loader:Loaded media_player from homeassistant.components.media_player
INFO:homeassistant.loader:Loaded lock from homeassistant.components.lock
INFO:homeassistant.loader:Loaded sensor from homeassistant.components.sensor
INFO:homeassistant.loader:Loaded switch from homeassistant.components.switch
INFO:homeassistant.loader:Loaded todo from homeassistant.components.todo
INFO:homeassistant.loader:Loaded vacuum from homeassistant.components.vacuum
INFO:homeassistant.setup:Setting up binary_sensor
INFO:homeassistant.components.binary_sensor:Setting up synthetic_home.binary_sensor
INFO:homeassistant.helpers.entity_registry:Registered new binary_sensor.synthetic_home entity: binary_sensor.living_room_motion_sensor
INFO:homeassistant.helpers.entity_registry:Registered new binary_sensor.synthetic_home entity: binary_sensor.living_room_motion_sensor_battery
INFO:homeassistant.helpers.entity_registry:Registered new binary_sensor.synthetic_home entity: binary_sensor.backyard_camera_motion
INFO:homeassistant.helpers.entity_registry:Registered new binary_sensor.synthetic_home entity: binary_sensor.backyard_camera_person
INFO:homeassistant.helpers.entity_registry:Registered new binary_sensor.synthetic_home entity: binary_sensor.backyard_camera_sound
INFO:homeassistant.helpers.entity_registry:Registered new binary_sensor.synthetic_home entity: binary_sensor.smart_lock
INFO:homeassistant.helpers.entity_registry:Registered new binary_sensor.synthetic_home entity: binary_sensor.smart_lock_tamper
INFO:homeassistant.helpers.entity_registry:Registered new binary_sensor.synthetic_home entity: binary_sensor.smart_lock_battery
INFO:homeassistant.setup:Setting up climate
INFO:homeassistant.components.climate:Setting up synthetic_home.climate
INFO:homeassistant.helpers.entity_registry:Registered new climate.synthetic_home entity: climate.thermostat
INFO:homeassistant.setup:Setting up fan
INFO:homeassistant.components.fan:Setting up synthetic_home.fan
INFO:homeassistant.setup:Setting up light
INFO:homeassistant.loader:Loaded weather from homeassistant.components.weather
INFO:homeassistant.loader:Loaded valve from homeassistant.components.valve
INFO:homeassistant.components.light:Setting up synthetic_home.light
INFO:homeassistant.helpers.entity_registry:Registered new light.synthetic_home entity: light.kitchen_light
INFO:homeassistant.helpers.entity_registry:Registered new light.synthetic_home entity: light.living_room_light
INFO:homeassistant.helpers.entity_registry:Registered new light.synthetic_home entity: light.game_room_light
INFO:homeassistant.helpers.entity_registry:Registered new light.synthetic_home entity: light.backyard_light
INFO:homeassistant.helpers.entity_registry:Registered new light.synthetic_home entity: light.garage_door_opener
INFO:homeassistant.helpers.entity_registry:Registered new light.synthetic_home entity: light.bedroom_1_light
INFO:homeassistant.helpers.entity_registry:Registered new light.synthetic_home entity: light.bedroom_2_light
INFO:homeassistant.helpers.entity_registry:Registered new light.synthetic_home entity: light.bedroom_3_light
INFO:homeassistant.helpers.entity_registry:Registered new light.synthetic_home entity: light.master_bedroom_light
INFO:homeassistant.setup:Setting up cover
INFO:homeassistant.components.cover:Setting up synthetic_home.cover
INFO:homeassistant.helpers.entity_registry:Registered new cover.synthetic_home entity: cover.garage_door_opener
INFO:homeassistant.helpers.entity_registry:Registered new cover.synthetic_home entity: cover.master_bedroom_smart_blinds
INFO:homeassistant.setup:Setting up lock
INFO:homeassistant.components.lock:Setting up synthetic_home.lock
INFO:homeassistant.helpers.entity_registry:Registered new lock.synthetic_home entity: lock.smart_lock
INFO:homeassistant.setup:Setting up media_player
INFO:homeassistant.components.media_player:Setting up synthetic_home.media_player
INFO:homeassistant.helpers.entity_registry:Registered new media_player.synthetic_home entity: media_player.smart_speaker
INFO:homeassistant.setup:Setting up sensor
INFO:homeassistant.components.sensor:Setting up synthetic_home.sensor
INFO:homeassistant.helpers.entity_registry:Registered new sensor.synthetic_home entity: sensor.thermostat_temperature
INFO:homeassistant.helpers.entity_registry:Registered new sensor.synthetic_home entity: sensor.thermostat_humidity
INFO:homeassistant.helpers.entity_registry:Registered new sensor.synthetic_home entity: sensor.living_room_motion_sensor_battery
INFO:homeassistant.helpers.entity_registry:Registered new sensor.synthetic_home entity: sensor.smart_lock_battery
INFO:homeassistant.helpers.entity_registry:Registered new sensor.synthetic_home entity: sensor.master_bedroom_smart_blinds_battery
INFO:homeassistant.setup:Setting up switch
INFO:homeassistant.components.switch:Setting up synthetic_home.switch
INFO:homeassistant.setup:Setting up todo
INFO:homeassistant.components.todo:Setting up synthetic_home.todo
INFO:homeassistant.setup:Setting up vacuum
INFO:homeassistant.components.vacuum:Setting up synthetic_home.vacuum
INFO:homeassistant.helpers.entity_registry:Registered new vacuum.synthetic_home entity: vacuum.roborock_downstairs
INFO:homeassistant.setup:Setting up weather
INFO:homeassistant.components.weather:Setting up synthetic_home.weather
INFO:homeassistant.setup:Setting up valve
INFO:homeassistant.components.valve:Setting up synthetic_home.valve
INFO:homeassistant.loader:Loaded ollama from homeassistant.components.ollama
INFO:homeassistant.setup:Setting up ollama
--------------------------------------------------------------- Captured log setup ----------------------------------------------------------------
INFO     homeassistant.loader:loader.py:779 Loaded synthetic_home from custom_components.synthetic_home
WARNING  homeassistant.loader:loader.py:694 We found a custom integration synthetic_home which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
INFO     homeassistant.loader:loader.py:779 Loaded homeassistant from homeassistant.components.homeassistant
INFO     homeassistant.setup:setup.py:384 Setting up homeassistant
INFO     homeassistant.loader:loader.py:779 Loaded conversation from homeassistant.components.conversation
INFO     homeassistant.loader:loader.py:779 Loaded http from homeassistant.components.http
INFO     homeassistant.loader:loader.py:779 Loaded intent from homeassistant.components.intent
INFO     homeassistant.setup:setup.py:384 Setting up http
WARNING  aiohttp_fast_zlib:__init__.py:39 zlib_ng and isal are not available, falling back to zlib, performance will be degraded.
INFO     homeassistant.setup:setup.py:384 Setting up intent
INFO     homeassistant.setup:setup.py:384 Setting up conversation
INFO     homeassistant.setup:setup.py:384 Setting up synthetic_home
INFO     custom_components.synthetic_home.model:model.py:140 value=['climate.ClimateEntityFeature.FAN_MODE', 'climate.ClimateEntityFeature.TURN_ON', 'climate.ClimateEntityFeature.TURN_OFF', 'climate.ClimateEntityFeature.TARGET_TEMPERATURE_RANGE'] (type=<class 'list'>)
INFO     custom_components.synthetic_home.model:model.py:140 value=['vacuum.VacuumEntityFeature.START', 'vacuum.VacuumEntityFeature.STOP', 'vacuum.VacuumEntityFeature.STATE', 'vacuum.VacuumEntityFeature.PAUSE', 'vacuum.VacuumEntityFeature.RETURN_HOME'] (type=<class 'list'>)
INFO     custom_components.synthetic_home.model:model.py:140 value=['media_player.MediaPlayerEntityFeature.PLAY', 'media_player.MediaPlayerEntityFeature.PAUSE', 'media_player.MediaPlayerEntityFeature.STOP', 'media_player.MediaPlayerEntityFeature.VOLUME_STEP', 'media_player.MediaPlayerEntityFeature.VOLUME_MUTE', 'media_player.MediaPlayerEntityFeature.VOLUME_SET', 'media_player.MediaPlayerEntityFeature.TURN_ON', 'media_player.MediaPlayerEntityFeature.TURN_OFF', 'media_player.MediaPlayerEntityFeature.NEXT_TRACK', 'media_player.MediaPlayerEntityFeature.PREVIOUS_TRACK'] (type=<class 'list'>)
INFO     custom_components.synthetic_home.model:model.py:140 value=['cover.CoverEntityFeature.OPEN', 'cover.CoverEntityFeature.CLOSE'] (type=<class 'list'>)
INFO     custom_components.synthetic_home.model:model.py:140 value=['cover.CoverEntityFeature.OPEN', 'cover.CoverEntityFeature.CLOSE', 'cover.CoverEntityFeature.SET_POSITION'] (type=<class 'list'>)
INFO     homeassistant.loader:loader.py:779 Loaded binary_sensor from homeassistant.components.binary_sensor
INFO     homeassistant.loader:loader.py:779 Loaded climate from homeassistant.components.climate
INFO     homeassistant.loader:loader.py:779 Loaded fan from homeassistant.components.fan
INFO     homeassistant.loader:loader.py:779 Loaded light from homeassistant.components.light
INFO     homeassistant.loader:loader.py:779 Loaded cover from homeassistant.components.cover
INFO     homeassistant.loader:loader.py:779 Loaded media_player from homeassistant.components.media_player
INFO     homeassistant.loader:loader.py:779 Loaded lock from homeassistant.components.lock
INFO     homeassistant.loader:loader.py:779 Loaded sensor from homeassistant.components.sensor
INFO     homeassistant.loader:loader.py:779 Loaded switch from homeassistant.components.switch
INFO     homeassistant.loader:loader.py:779 Loaded todo from homeassistant.components.todo
INFO     homeassistant.loader:loader.py:779 Loaded vacuum from homeassistant.components.vacuum
INFO     homeassistant.setup:setup.py:384 Setting up binary_sensor
INFO     homeassistant.components.binary_sensor:entity_platform.py:346 Setting up synthetic_home.binary_sensor
INFO     homeassistant.helpers.entity_registry:entity_registry.py:912 Registered new binary_sensor.synthetic_home entity: binary_sensor.living_room_motion_sensor
INFO     homeassistant.helpers.entity_registry:entity_registry.py:912 Registered new binary_sensor.synthetic_home entity: binary_sensor.living_room_motion_sensor_battery
INFO     homeassistant.helpers.entity_registry:entity_registry.py:912 Registered new binary_sensor.synthetic_home entity: binary_sensor.backyard_camera_motion
INFO     homeassistant.helpers.entity_registry:entity_registry.py:912 Registered new binary_sensor.synthetic_home entity: binary_sensor.backyard_camera_person
INFO     homeassistant.helpers.entity_registry:entity_registry.py:912 Registered new binary_sensor.synthetic_home entity: binary_sensor.backyard_camera_sound
INFO     homeassistant.helpers.entity_registry:entity_registry.py:912 Registered new binary_sensor.synthetic_home entity: binary_sensor.smart_lock
INFO     homeassistant.helpers.entity_registry:entity_registry.py:912 Registered new binary_sensor.synthetic_home entity: binary_sensor.smart_lock_tamper
INFO     homeassistant.helpers.entity_registry:entity_registry.py:912 Registered new binary_sensor.synthetic_home entity: binary_sensor.smart_lock_battery
INFO     homeassistant.setup:setup.py:384 Setting up climate
INFO     homeassistant.components.climate:entity_platform.py:346 Setting up synthetic_home.climate
INFO     homeassistant.helpers.entity_registry:entity_registry.py:912 Registered new climate.synthetic_home entity: climate.thermostat
INFO     homeassistant.setup:setup.py:384 Setting up fan
INFO     homeassistant.components.fan:entity_platform.py:346 Setting up synthetic_home.fan
INFO     homeassistant.setup:setup.py:384 Setting up light
INFO     homeassistant.loader:loader.py:779 Loaded weather from homeassistant.components.weather
INFO     homeassistant.loader:loader.py:779 Loaded valve from homeassistant.components.valve
INFO     homeassistant.components.light:entity_platform.py:346 Setting up synthetic_home.light
INFO     homeassistant.helpers.entity_registry:entity_registry.py:912 Registered new light.synthetic_home entity: light.kitchen_light
INFO     homeassistant.helpers.entity_registry:entity_registry.py:912 Registered new light.synthetic_home entity: light.living_room_light
INFO     homeassistant.helpers.entity_registry:entity_registry.py:912 Registered new light.synthetic_home entity: light.game_room_light
INFO     homeassistant.helpers.entity_registry:entity_registry.py:912 Registered new light.synthetic_home entity: light.backyard_light
INFO     homeassistant.helpers.entity_registry:entity_registry.py:912 Registered new light.synthetic_home entity: light.garage_door_opener
INFO     homeassistant.helpers.entity_registry:entity_registry.py:912 Registered new light.synthetic_home entity: light.bedroom_1_light
INFO     homeassistant.helpers.entity_registry:entity_registry.py:912 Registered new light.synthetic_home entity: light.bedroom_2_light
INFO     homeassistant.helpers.entity_registry:entity_registry.py:912 Registered new light.synthetic_home entity: light.bedroom_3_light
INFO     homeassistant.helpers.entity_registry:entity_registry.py:912 Registered new light.synthetic_home entity: light.master_bedroom_light
INFO     homeassistant.setup:setup.py:384 Setting up cover
INFO     homeassistant.components.cover:entity_platform.py:346 Setting up synthetic_home.cover
INFO     homeassistant.helpers.entity_registry:entity_registry.py:912 Registered new cover.synthetic_home entity: cover.garage_door_opener
INFO     homeassistant.helpers.entity_registry:entity_registry.py:912 Registered new cover.synthetic_home entity: cover.master_bedroom_smart_blinds
INFO     homeassistant.setup:setup.py:384 Setting up lock
INFO     homeassistant.components.lock:entity_platform.py:346 Setting up synthetic_home.lock
INFO     homeassistant.helpers.entity_registry:entity_registry.py:912 Registered new lock.synthetic_home entity: lock.smart_lock
INFO     homeassistant.setup:setup.py:384 Setting up media_player
INFO     homeassistant.components.media_player:entity_platform.py:346 Setting up synthetic_home.media_player
INFO     homeassistant.helpers.entity_registry:entity_registry.py:912 Registered new media_player.synthetic_home entity: media_player.smart_speaker
INFO     homeassistant.setup:setup.py:384 Setting up sensor
INFO     homeassistant.components.sensor:entity_platform.py:346 Setting up synthetic_home.sensor
INFO     homeassistant.helpers.entity_registry:entity_registry.py:912 Registered new sensor.synthetic_home entity: sensor.thermostat_temperature
INFO     homeassistant.helpers.entity_registry:entity_registry.py:912 Registered new sensor.synthetic_home entity: sensor.thermostat_humidity
INFO     homeassistant.helpers.entity_registry:entity_registry.py:912 Registered new sensor.synthetic_home entity: sensor.living_room_motion_sensor_battery
INFO     homeassistant.helpers.entity_registry:entity_registry.py:912 Registered new sensor.synthetic_home entity: sensor.smart_lock_battery
INFO     homeassistant.helpers.entity_registry:entity_registry.py:912 Registered new sensor.synthetic_home entity: sensor.master_bedroom_smart_blinds_battery
INFO     homeassistant.setup:setup.py:384 Setting up switch
INFO     homeassistant.components.switch:entity_platform.py:346 Setting up synthetic_home.switch
INFO     homeassistant.setup:setup.py:384 Setting up todo
INFO     homeassistant.components.todo:entity_platform.py:346 Setting up synthetic_home.todo
INFO     homeassistant.setup:setup.py:384 Setting up vacuum
INFO     homeassistant.components.vacuum:entity_platform.py:346 Setting up synthetic_home.vacuum
INFO     homeassistant.helpers.entity_registry:entity_registry.py:912 Registered new vacuum.synthetic_home entity: vacuum.roborock_downstairs
INFO     homeassistant.setup:setup.py:384 Setting up weather
INFO     homeassistant.components.weather:entity_platform.py:346 Setting up synthetic_home.weather
INFO     homeassistant.setup:setup.py:384 Setting up valve
INFO     homeassistant.components.valve:entity_platform.py:346 Setting up synthetic_home.valve
INFO     homeassistant.loader:loader.py:779 Loaded ollama from homeassistant.components.ollama
INFO     homeassistant.setup:setup.py:384 Setting up ollama
```

</details> 

## What I've done to get to this point

I'm trying to run the evaluations using ollama and llama3.2:1b following the instructions in https://github.com/allenporter/home-assistant-datasets/blob/32607a348c48af3440e67098a21615816310fdde/datasets/assist/README.md

I've got the environment set up correctly, I think. I took the following steps:
1. Open up this repo in a dev container
2. Use pip3 to install the requirements_dev and requirements_eval
3. Add [home-assistant-synthetic-home](https://github.com/allenporter/home-assistant-synthetic-home) to PYTHONPATH
4. Add a `secrets.yaml` to the repository root. Because I'm just using ollama, I just need to give the correct URL for its API:
    ```yaml
    google_api_key: XXXXXXXXXXXXXXXXXXXX
    openai_api_key: sk-XXXXXXXXXXXXXXXXXXXX
    anthropic_api_key: ant-XXXXXXXXXXXXXXXXXXXX
    vicuna_convesation_base_url: http://llama-cublas.llama:8000/v1
    ollama_url: http://127.0.0.1:11434
    ```
6. Export the DATASET, OUTPUT_DIR and MODEL env vars and run the home-assistant-datasets command with them:
    ```bash
    DATASET="datasets/assist/"
    OUTPUT_DIR="reports/assist/2024.unknown"
    MODEL=llama3.2-1b
    home-assistant-datasets assist collect --models=${MODEL} --dataset=${DATASET} --model_output_dir=${OUTPUT_DIR}
    ```

## Troubleshooting steps I've taken

I _think_ my Python environment is set up correctly. The error doesn't seem to be complaining about dependencies, and everything installed happily in the devcontainer.

As above, the actual state that it seems to get is `ConfigEntryState.SETUP_RETRY`. I looked this up and found [this page](https://developers.home-assistant.io/docs/config_entries_index/) which suggested a dependency of the config entry is not ready, but I'm not sure which one that would be.

Another idea I had, based on the retry wording was just to wait longer. I tried adding a 30s wait before the assertion but this did not fix it either.

Ollama appears to work fine locally so I don't think it's a problem here. E.g. I can hit it with curl and get back working responses, or use `ollama run llama3.2:1b` successfully.

## Misc

Apologies if this is the wrong place for this query! Happy to redirect it elsewhere if so.

I should also flag that I'm a novice in the realm of home assistant programming, so there may well be something obvious I've missed.

Thanks for creating such an awesome project!

**Comment by allenporter** [comment]
Hi, you may want to add '-- -vv' to get more details about the log failure.  I suspect this means ollama failed to set up

**Comment by domdomegg** [comment]
**tldr: I set my `ollama_url` wrong. It should have pointed at the docker host, not localhost, given I'm running Docker on Mac.**

Thank you! Enabling the verbose logs (at DEBUG level) helped me identify it was hitting a connection failure to Ollama from within the devcontainer.

I forgot that because I'm running on a Mac with colima, Docker is actually running in a virtual machine. Therefore my devcontainer which is running in Docker is running on that virtual machine, which isn't where Ollama is running. I needed to set the ollama_url to the `docker.internal.host` (or `http://192.168.5.2:11434` specifically for colima, with the Ollama port).

This has now enabled me to run the evaluation. I might create a YouTube video tutorial explaining how I did all this because I think this project is awesome, but can be confusing to set up (at least for a person less familiar with it all like me :)).

Thanks again for your help.
